Add file cache functionality

jmp1985 opened the pull request:
This pull request adds a file cache. When a file is requested, first a cache folder (default ~/.cache/pdb) is checked. If the file exists either as a ${UNIPROT_ID}.cif or ${UNIPROT_ID}.pdb file, then that file is returned. Otherwise the file is downloaded.

The cache is wrapped in a class within profet.cache.PDBFileCache. You can set the cache programmatically by setting the save_directory in profet.

mooniean commented on profet/pdb.py:
```diff
@@ -79,12 +76,5 @@ def get_pdb(
             )
 
-        # Update the filename
-        filedir = filedir + "_" + pdb_id + "." + filetype
```
this reverts back from the saving with pdbID when downloaded from pdb. Can we cache that too?

jmp1985 replied:
Ah, sorry I didn't realise I had reverted something... I was obviously too hasty with my changes! I will address this now and update the pull request.

mooniean reviewed (commented):
This is all okay, just a comment about saving with pdbID when downloading from PDB

jmp1985 commented:
Hi @mooniean 

I've updated the pull request. I've modified it slightly to only add the pdb id when it is different from the uniprot_id. Is that the correct functionality. 

Perhaps we could add a test to ensure that what is implemented matches what we want to happen.

Best wishes
James

mooniean commented:
> Hi @mooniean
> 
> I've updated the pull request. I've modified it slightly to only add the pdb id when it is different from the uniprot_id. Is that the correct functionality.
> 
> Perhaps we could add a test to ensure that what is implemented matches what we want to happen.


Yes, that's a good solution! We added it so we know which of the PDBs was downloaded when several are available. 

As for the test, I'll make it an issue so we don't forget!

mooniean reviewed (approved):
LGTM!

NikJur reviewed (approved):
Nice. Looks good to me.

jmp1985 commented:
Regarding Issue #33 I have added a test for the cache.
